### ISSUE TITLE
Newsbeuter exits after keeping it running for over 3 days

### ISSUE THREAD

**Issue by abhixec** [akrennmair/newsbeuter#462]
**Newsbeuter version (copy from `newsbeuter -v`)**: 2.9

Steps to reproduce the issue:

1. Open newsbeuter, keep it running for few days
2. After few days when I go back to the window that had newsbeuter, all i saw was the app closed and there was some malloc message (the spacing was off and had random characters)  so couldn't get a clean log.

Is there a way to check for the logs? 
**Other info you think is relevant**:
I have auto reload set not sure if that would be helpful in reproducing the problem.
Ran newsbeuter on a tmux session

**Comment by Minoru** [comment]
>2. After few days when I go back to the window that had newsbeuter, all
>i saw was the app closed and there was some malloc message (the spacing
>was off and had random characters)  so couldn't get a clean log.

A screenshot will do.

>Is there a way to check for the logs?

Newsbeuter doesn't normally keep a log. You have to run it with the following options: -l6 -dnewsbeuter.log

This will produce a log file in the current directory. It will be quite large, but I'm only interested in the last 20 lines or so.

>I have auto reload set not sure if that would be helpful in reproducing
>the problem.

Probably. Can you post your config file?

You can also try and increase the frequency of updates - that should make Newsbeuter crash more often.

**Comment by abhixec** [comment]
config
```
auto-reload yes
notify-xterm yes
browser "open -a /Applications/FirefoxDeveloperEdition.app/Contents/MacOS/firefox"
always-display-description true
```
urls
```
http://talkchelsea.net/feed
https://news.ycombinator.com/rss
http://www.engadget.com/rss.xml
http://feeds.feedburner.com/TechCrunch/
http://rss.slashdot.org/Slashdot/slashdot
http://feeds.feedburner.com/d0od
https://vivaldi.net/?format=feed&type=rss
http://news.yahoo.com/rss/business
http://news.yahoo.com/rss/internet
http://feeds.feedburner.com/AndroidPolice
http://feed.androidauthority.com/
#https://www.technologyreview.com/stories.rss
#https://www.technologyreview.com/c/computing/rss/

```

**Comment by Minoru** [comment]
@abhixec, how is it going? Were you able to reproduce the crash?

**Comment by der-lyse** [comment]
I probably experienced the very same bug, too. For two weeks now (started on 2017-01-27) I'm running newsbeuter with debug level 6, but no crash was encountered so far. It looks like the debug mode masks the problem somehow, because "usually" the crash had happened after just a couple of days.

Currently the log file is about 2.28 million lines long and around 240 MiB in size. I'll keep you posted.

**Comment by Minoru** [comment]
@der-lyse, two weeks is quite long for an experiment! You might want to switch tactics: enable core dumps (`ulimit -c unlimited`) and run Newsbeuter with logging disabled. Coredump isn't the same as a log, of course, but perhaps it'll give us some info nonetheless.

**Comment by der-lyse** [comment]
@Minoru: Alright. Let's see.

**Comment by der-lyse** [comment]
Well, well, well, it's still running without any crashes so far.

**Comment by abhixec** [comment]
I have been running newsbeuter for roughly more than 2 weeks didn't encounter this issue but the memory usage spiked to 11G (is that expected?)

**Comment by Minoru** [comment]
> the memory usage spiked to 11G (is that expected?)

With 2.9, yes. We has a memory leak which is now fixed.

**Comment by der-lyse** [comment]
Time for a tiny update: I've been running a version built from commit 6c1dc60d68be4adfc8dcf9dee5f08fc8b21d7362 for quite some time now and didn't experience segfaults anymore so far. The bug we encountered may have been fixed in the meantime. Sadly I cannot remember exactly which version I was using when the crashes occurred, perhaps 2.8.

(I restarted newsbeuter several times in the past two weeks due to feed management, so newsbeuter's uptime was always only a couple of days.)
